xpartition.py

```diff
@@ -214,21 +214,21 @@ def _collect_by_partition(
     DataArrays that can be written out to those partitions.
     """
     dataarrays = collections.defaultdict(list)
-    for da in ds.data_vars.values():
+    for da in {**ds.coords, **ds.data_vars}.values():
```

Comment:
nit: I've always found iterating over xarray.Dataset confusing (i.e. not sure what exactly will be included in that iterable). Anyway I think this is good for being explicit about including `coords`. But is there some way to get the names or values of the coords + data_vars in a single call?

Reply:
No, I think what you have makes sense, especially if the alternative is more awkward for getting the DataArrays.